src/overlays/actors/ovl_Arms_Hook/z_arms_hook.c

```diff
@@ -171,10 +172,11 @@ void ArmsHook_Shoot(ArmsHook* this, GlobalContext* globalCtx) {
     if ((this->timer != 0) && (this->collider.base.atFlags & AT_HIT) &&
         (this->collider.info.atHitInfo->elemType != ELEMTYPE_UNK4)) {
         touchedActor = this->collider.base.at;
-        if ((touchedActor->update != NULL) && (touchedActor->flags & (ACTOR_FLAG_9 | ACTOR_FLAG_10))) {
+        if ((touchedActor->update != NULL) &&
+            (touchedActor->flags & (ACTOR_FLAG_HOOK_CAN_CARRY | ACTOR_FLAG_HOOK_BRING_PLAYER))) {
             if (this->collider.info.atHitInfo->bumperFlags & BUMP_HOOKABLE) {
                 ArmsHook_AttachHookToActor(this, touchedActor);
-                if (CHECK_FLAG_ALL(touchedActor->flags, ACTOR_FLAG_10)) {
+                if (CHECK_FLAG_ALL(touchedActor->flags, ACTOR_FLAG_HOOK_BRING_PLAYER)) {
                     func_80865044(this);
```

Comment:
Again, relying on undocumented code makes it hard to understand the flags

But I can't say I understand even the documented parts of Hookshot anyway

`ArmsHook#child != NULL` seems to indicate pulling player to wherever hookshot is

For pulling something to player I can't figure out how it works, seems to pull in the direction `grabbedDistDiff` holds, but that holds the hookshot to hooked thing difference, which sounds like it would be different from the player to hookshot direction in many cases (for example hitting the side of an object with hookshot)

Reply:
yeah im not sure on the exact implementation details of how the hookshot pulls the player. but based on context and usage i think its safe to say that this is correct. `func_80865044` which does the child/parent handling gets called if `SurfaceType_IsHookshotSurface` passes, which is how general hookshottable collision that pulls player is handled. and if an actor has this flag set it also calls `func_80865044` 

plenty of other context shows that this is its usage, ganondorf enables it when hes vulnerable and the player can hookshot over to him for example. The scarecrows enable this by default etc

Reply:
and ofc tested in game
https://user-images.githubusercontent.com/22045047/151274339-4daaeba5-c949-4bd0-9000-f5db3a0226d0.mp4